File: test-baru-lagi/soal3/main.py
def removeSpaces(inputStr):
    return input.replace(" ","")

def changetolowercase(input):
    return input.lower()

def countCharacters(inputStr):
    pecahtext = {}
    for char in inputStr:
        if char in pecahtext:
            pecahtext[char] += 1
        else:
            pecahtext[char] = 1
    return pecahtext
     
def printCharacterCounts(inputStr):
    counts = countCharacters(inputStr)
    outputStr = ""
    for char in counts:
        count = counts[char]
        if count == 1:
            outputStr += char
        else:
            outputStr += str(count) + char
    return outputStr

def main(input):
    return printCharacterCounts(removeSpaces(changetolowercase(input)))


input = "dani Maulana"
print("text yang anda masukkan adalah: "+ input);
print(countCharacters(input))
output = main(input)
print("Hasil setelah di kelola: "+ output);
print("")

File: test-baru-lagi/soal3/main.go
package main

import (
	"fmt"
	"strings"
)

func removeSpaces(inputStr string) string {
	return strings.ReplaceAll(inputStr, " ", "")
}

func changeToLowercase(input string) string {
	return strings.ToLower(input)
}

func countCharacters(inputStr string) map[string]int {
	pecahtext := make(map[string]int)
	for _, char := range inputStr {
		charStr := string(char)
		if _, ok := pecahtext[charStr]; ok {
			pecahtext[charStr] += 1
		} else {
			pecahtext[charStr] = 1
		}
	}
	return pecahtext
}

func printCharacterCounts(inputStr string) string {
	counts := countCharacters(inputStr)
	var outputStr string
	for char, count := range counts {
		if count == 1 {
			outputStr += char
		} else {
			outputStr += fmt.Sprintf("%d%s", count, char)
		}
	}
	return outputStr
}

func main() {
	input := "dani Maulana"
	fmt.Println(printCharacterCounts(removeSpaces(changeToLowercase(input))))
}
